merino/providers/suggest/sports/provider.py

```
@@ -0,0 +1,142 @@
+                # and collate by sport. Product and UI do not want that, so
+                # we will always return only one sport.
+                results = await self.backend.query(sreq.query, score=self.score, url=self.url)
+                return [
+                    self.build_suggestion(
+                        # If we are collating, this should be the sport group.
+                        sport_name="all",
+                        query=sreq.query,
+                        # This should be the max es_score value returned to
+                        # act as a general score adjustment for this suggestion
+                        score_adj=0,
+                        events=results,
+                    )
+                    for result in results
+                ]
+        return []
+
+    def normalize_query(self, query: str) -> str:
+        """Perform whatever steps are required to normalize the user provided query string"""
+        return super().normalize_query(query)
+
+    def validate(self, srequest: SuggestionRequest) -> None:
+        """Validate the incoming request"""
+        return super().validate(srequest)
+
+    def build_suggestion(
+        self,
+        sport_name: str,
+        query: str,
+        events: list[SportSummary],
+        score_adj: float = 0,
+    ) -> BaseSuggestion:
+        """Build a base suggestion with the sport data results"""
+        # "All" returns all sport events that match the search criteria.
+        # Further work would be required to return results collated to their
+        # respective sports. Each would be returned based on the `SportSummary.sport`.
+        if sport_name == "all":
+            sport_name = "All Sport"
+            details = SportEventDetails(events[0])
+        else:
+            # Return an error because we shouldn't currently be returning anything
+            # other than "mixed" results.
+            raise SportsDataError("Multiple Sports Returned")
```

Looks like exceptions are not handled by the caller (`query()`) and Merino will return a 500 error in this case, is that expected? Normally, for those internal issues, Merino would return an empty suggestion list to the client and log the error(s) with some metrics instead.